pkg/traefik/errors.go

```diff
@@ -0,0 +1,50 @@
+package traefik
+
+import (
+	"fmt"
+)
+
+type Error struct {
+	Code    string
+	Message string
+	Err     error
+	Params  []interface{}
+}
+
+func (e *Error) Error() string {
+	msg := fmt.Sprintf(e.Message, e.Params...)
+	if e.Err != nil {
+		return fmt.Sprintf("%s: %v", msg, e.Err)
+	}
+	return msg
+}
+
+func (e *Error) Wrap(err error) error {
+	e.Err = err
+	return e
+}
```

Comment:
This isn't wrapping, this is replacing. Shouldn't it be

```suggestion
func (e *Error) Wrap(err error) error {
	e.Err = errors.Join(e.Err, err)
	return e
}
```

Reply:
very neat. thanks man